diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8e5383ac98..cc569eee3a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1031,6 +1031,9 @@ if(USE_NGHTTP2)
   find_package(NGHTTP2)
   if(NGHTTP2_FOUND)
     list(APPEND CURL_LIBS CURL::nghttp2)
+  elseif("${CURL_CA_BUNDLE}" STREQUAL "env")
+    unset(CURL_CA_BUNDLE CACHE)
+    set(CURL_ENABLE_CA_BUNDLE_ENVVAR)
   else()
     set(USE_NGHTTP2 OFF)
   endif()
@@ -1314,7 +1317,7 @@ endif()
 # CA handling
 #
 if(_curl_ca_bundle_supported)
-  set(_ca_opt_desc "Set 'none' to disable or 'auto' for auto-detection. Defaults to 'auto'.")
+  set(_ca_opt_desc "Set 'none' to disable, 'auto' for auto-detection at build time, or 'env' for runtime setting from CURL_CA_BUNDLE in environment.. Defaults to 'auto'.")
 
   set(CURL_CA_BUNDLE "auto" CACHE
     STRING "Absolute path to the CA bundle. ${_ca_opt_desc}")
@@ -1330,7 +1333,7 @@ if(_curl_ca_bundle_supported)
   endif()
 
   if(CURL_CA_BUNDLE STREQUAL "")
-    message(FATAL_ERROR "Invalid value of CURL_CA_BUNDLE. Use 'none', 'auto' or file path.")
+    message(FATAL_ERROR "Invalid value of CURL_CA_BUNDLE. Use 'none', 'auto', 'env' or file path.")
   elseif(CURL_CA_BUNDLE STREQUAL "none")
     unset(CURL_CA_BUNDLE CACHE)
   elseif(CURL_CA_BUNDLE STREQUAL "auto")
diff --git a/acinclude.m4 b/acinclude.m4
index 775102033c..9779c8826d 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -1213,7 +1213,7 @@ AS_HELP_STRING([--without-ca-path], [Do not use a default CA path]),
     fi
   fi
 
-  if test "x$ca" = "xno" || test -f "$ca"; then
+  if test "x$ca" = "xno" || test "x$ca" = "xenv" || test -f "$ca"; then
     ca_warning=""
   fi
 
@@ -1237,7 +1237,11 @@ AS_HELP_STRING([--without-ca-path], [Do not use a default CA path]),
     capath_warning=""
   fi
 
-  if test "x$ca" != "xno"; then
+  if test "x$ca" = "xenv"; then
+    AC_DEFINE_UNQUOTED(CURL_ENABLE_CA_BUNDLE_ENVVAR, 1,
+      [Define to 1 to support loading CURL_CA_BUNDLE from env at runtime.])
+    AC_MSG_RESULT([use environment at runtime])
+  elif test "x$ca" != "xno"; then
     CURL_CA_BUNDLE="$ca"
     AC_DEFINE_UNQUOTED(CURL_CA_BUNDLE, "$ca", [Location of default ca bundle])
     AC_SUBST(CURL_CA_BUNDLE)
diff --git a/lib/curl_config-cmake.h.in b/lib/curl_config-cmake.h.in
index d8d9d7ecbe..faba162415 100644
--- a/lib/curl_config-cmake.h.in
+++ b/lib/curl_config-cmake.h.in
@@ -25,6 +25,9 @@
 /* Location of default ca bundle */
 #cmakedefine CURL_CA_BUNDLE "${CURL_CA_BUNDLE}"
 
+/* Whether to enable specifying CURL_CA_BUNDLE in environment */
+#cmakedefine CURL_ENABLE_CA_BUNDLE_ENVVAR 1
+
 /* define "1" to use built-in ca store of TLS backend */
 #cmakedefine CURL_CA_FALLBACK 1
 
diff --git a/lib/vtls/vtls.c b/lib/vtls/vtls.c
index be0c032bff..66cc2e8ee3 100644
--- a/lib/vtls/vtls.c
+++ b/lib/vtls/vtls.c
@@ -289,7 +289,7 @@ static void free_primary_ssl_config(struct ssl_primary_config *sslc)
 CURLcode Curl_ssl_easy_config_complete(struct Curl_easy *data)
 {
   struct ssl_config_data *sslc = &data->set.ssl;
-#if defined(CURL_CA_PATH) || defined(CURL_CA_BUNDLE)
+#if defined(CURL_CA_PATH) || defined(CURL_CA_BUNDLE) || defined(CURL_ENABLE_CA_BUNDLE_ENVVAR)
   struct UserDefined *set = &data->set;
   CURLcode result;
 #endif
@@ -306,7 +306,21 @@ CURLcode Curl_ssl_easy_config_complete(struct Curl_easy *data)
         return result;
     }
 #endif
-#ifdef CURL_CA_BUNDLE
+#if defined(CURL_ENABLE_CA_BUNDLE_ENVVAR)
+    char *env;
+    env = curl_getenv("CURL_CA_BUNDLE");
+    if(env) {
+      result = Curl_setstropt(&set->str[STRING_SSL_CAFILE], env);
+      if (result)
+        return result;
+
+      result = Curl_setstropt(&set->str[STRING_SSL_CAFILE_PROXY], env);
+      if (result)
+        return result;
+
+      curl_free(env);
+    }
+#elif defined(CURL_CA_BUNDLE)
     if(!sslc->custom_cafile && !set->str[STRING_SSL_CAFILE]) {
       result = Curl_setstropt(&set->str[STRING_SSL_CAFILE], CURL_CA_BUNDLE);
       if(result)
