diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 4339c2921..7de5f715c 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -1484,6 +1484,7 @@ config("default_warnings") {
       # Disables.
       "-Wno-missing-field-initializers",  # "struct foo f = {0};"
       "-Wno-unused-parameter",  # Unused function parameters.
+      "-Wno-invalid-offsetof", # Use of "conditionally-supported" offsetof in c++17
     ]
   }
 
@@ -1853,8 +1854,17 @@ config("no_incompatible_pointer_warnings") {
 # Shared settings for both "optimize" and "optimize_max" configs.
 # IMPORTANT: On Windows "/O1" and "/O2" must go before the common flags.
 if (is_win) {
-  common_optimize_on_cflags = [
-    "/Ob2",  # Both explicit and auto inlining.
+  common_optimize_on_cflags = []
+  if(is_clang) {
+    common_optimize_on_cflags += [
+     "/Ob2",  # Both explicit and auto inlining.
+    ]
+  } else {
+    common_optimize_on_cflags += [
+      "/Ob3",  # Both explicit and auto inlining.
+    ]
+  }
+  common_optimize_on_cflags += [
     "/Oy-",  # Disable omitting frame pointers, must be after /O2.
     "/Zc:inline",  # Remove unreferenced COMDAT (faster links).
   ]
